packages/editor/src/components/block-removal-warnings/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+
+import { __ } from '@wordpress/i18n';
+import { privateApis as blockEditorPrivateApis } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+
+const { BlockRemovalWarningModal } = unlock( blockEditorPrivateApis );
+
+// Prevent accidental removal of certain blocks, asking the user for confirmation first.
+const blockRemovalRules = {
+	'core/query': {
+		message: __(
+			'The Query Loop block displays a list of posts or pages, so removing it will prevent that content from displaying.'
+		),
+	},
+	'core/post-content': {
+		message: __(
+			'Removing the Post Content block will stop your post or page content from displaying on this template.'
+		),
+	},
```

Comment:
Are we showing this message for the removal of all query blocks?

Reply:
see reply above.